src/reader/parser.rs

```diff
@@ -349,15 +349,15 @@ impl PullParser {
     /// # Parameters
     /// * `t`       --- next token;
     /// * `on_name` --- a callback which is executed when whitespace is encountered.
-    fn read_qualified_name(&mut self, t: Token, target: QualifiedNameTarget,
-                           on_name: |&mut PullParser, Token, OwnedName| -> Option<XmlEvent>) -> Option<XmlEvent> {
+    fn read_qualified_name<'a, F>(&'a mut self, t: Token, target: QualifiedNameTarget, on_name: F) -> Option<XmlEvent> 
+      where F: Fn<(&'a mut PullParser, Token, OwnedName), Option<XmlEvent>> {
```

Comment:
Should be `where F: Fn(&mut PullParser, Token, OwnedName) -> Option<XmlEvent>`.
The `'a` lifetime is incorrect and should be removed (well, it may work, but it changes the semantic of the function).